astropy/time/formats.py

```diff
@@ -942,6 +942,31 @@ class TimeYearDayTime(TimeISO):
                 '{year:d}:{yday:03d}'))
 
 
+class TimeDatetime64(TimeISOT):
+    name = 'datetime64'
+    def _check_val_type(self, val1, val2):
+        # Note: don't care about val2 for this class`
+        if not all(isinstance(val, np.datetime64) for val in val1.flat):
+            raise TypeError('Input values for {0} class must be '
+                            'datetime64 objects'.format(self.name))
+        return val1, None
+
+    def set_jds(self, val1, val2):
+        if val1.dtype.name in ['datetime64[M]', 'datetime64[Y]']:
+            val1 = val1.astype('M8[D]')
+
+        val1 = val1.astype('<U30')
```

Comment:
You should be able to use `val1.astype('S')`.  This saves space and will just have as many characters as are needed.